praw/reddit.py

```diff
@@ -579,10 +646,25 @@ def post(
             None).
 
         """
-        data = self.request(
-            "POST", path, data=data or {}, files=files, params=params
-        )
-        return self._objector.objectify(data)
+        data = data or {}
+        try:
+            return self._objectify_request(
+                data=data,
+                files=files,
+                method="POST",
+                params=params,
+                path=path,
+            )
+        except RedditAPIException as exception:
+            return self._handle_rate_limit(
```

Comment:
I would like to see this a conditional that just says how many seconds to sleep. So the the retry or raise logic exists here.